sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -59,31 +65,120 @@ public HttpPipeline getHttpPipeline() {
+            .addWithResponseAsync(digitalTwinId, payload)
+            .flatMap(
+                response -> {
+                    try {
+                        String jsonResponse = mapper.writeValueAsString(response.getValue());
+                        return Mono.just(new SimpleResponse<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), jsonResponse));
+                    } catch (JsonProcessingException e) {
+                        return Mono.error(e);
+                    }
+                });
+    }
+
+    // Input is String and output is Response<String> -> ResponseBase<DigitalTwinsAddHeaders, String>.
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<ResponseBase<DigitalTwinsAddHeaders, String>> createDigitalTwinWithResponseResponseBaseString(String digitalTwinId, String digitalTwin) throws JsonProcessingException {
+        Object payload = mapper.readValue(digitalTwin, Object.class);
+        return protocolLayer
+            .getDigitalTwins()
+            .addWithResponseAsync(digitalTwinId, payload)
+            .flatMap(
+                response -> {
+                    try {
+                        String jsonResponse = mapper.writeValueAsString(response.getValue());
+                        return Mono.just(new ResponseBase<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), jsonResponse, response.getDeserializedHeaders()));
+                    } catch (JsonProcessingException e) {
+                        return Mono.error(e);
+                    }
+                });
+    }
+
+    // Input is String and output is Response<String> -> DigitalTwinsAddResponse (json string).
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DigitalTwinsAddResponse> createDigitalTwinWithResponseDigitalTwinAddResponseString(String digitalTwinId, String digitalTwin) throws JsonProcessingException {
+        Object payload = mapper.readValue(digitalTwin, Object.class);
+        return protocolLayer
+            .getDigitalTwins()
+            .addWithResponseAsync(digitalTwinId, payload)
+            .flatMap(
+                response -> {
+                    try {
+                        String jsonResponse = mapper.writeValueAsString(response.getValue());
+                        DigitalTwinsAddResponse addResponse = new DigitalTwinsAddResponse(response.getRequest(), response.getStatusCode(), response.getHeaders(), jsonResponse, response.getDeserializedHeaders());
+                        return Mono.just(addResponse);
+                    } catch (JsonProcessingException e) {
+                        return Mono.error(e);
+                    }
+                });
+    }
+
+    // Input is Object and output is Response<Object> -> DigitalTwinsAddResponse.
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DigitalTwinsAddResponse> createDigitalTwinWithResponseDigitalTwinsAddResponseObject(String digitalTwinId, Object digitalTwin) {
```

Comment:
Option 4:
Accept request as object and send to PL.
PL returns response as an object, return to user.

Note:
```java
// Convert to Jackson's tree model, which is useful to parse json string when you are not sure what the json string looks like
JsonNode jsonNode = mapper.valueToTree(result);

// Since the return type is defined as Object, but is dynamically returned as a Map, you can also do the following: (JsonNode approach is cleaner, though)
String createdTwinModelId = ((LinkedHashMap)((LinkedHashMap)result).get("$metadata")).get("$model").toString();
```

Response type returned is an implementation of `ResponseBase<H, T>` - it contains the etag info in explicitly defined `deserializedHeaders`.